microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/project/AbstractConfigSource.java

```diff
@@ -0,0 +1,167 @@
+/*******************************************************************************
+* Copyright (c) 2020 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package com.redhat.microprofile.jdt.internal.core.project;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.attribute.FileTime;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.jdt.core.IClasspathEntry;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.jdt.internal.core.JavaProject;
+
+/**
+ * Abstract class for config file.
+ * 
+ * @author Angelo ZERR
+ *
+ * @param <T> the config model (ex: Properties for *.properties file)
+ */
+public abstract class AbstractConfigSource<T> implements IConfigSource {
+
+	private static final Logger LOGGER = Logger.getLogger(AbstractConfigSource.class.getName());
+
+	private final String configFileName;
+	private final IJavaProject javaProject;
+	private Path configFile;
+
+	private FileTime lastModified;
+
+	private T config;
+
+	public AbstractConfigSource(String configFileName, IJavaProject javaProject) {
+		this.configFileName = configFileName;
+		this.javaProject = javaProject;
+	}
+
+	/**
+	 * Returns the target/classes/$configFile and null otherwise.
+	 * 
+	 * <p>
+	 * Using this file instead of using src/main/resources/$configFile gives the
+	 * capability to get the filtered value.
+	 * </p>
+	 * 
+	 * @return the target/classes/$configFile and null otherwise.
+	 */
+	private Path getConfigFile() {
+		if (configFile != null && Files.exists(configFile)) {
+			return configFile;
+		}
+		try {

```

Comment:
make sure IJavaProject (getProject) is still accessible

Reply:
@fbricon updated